src/main/java/io/jenkins/plugins/oidc_provider/IdTokenCredentials.java

```diff
@@ -161,4 +180,75 @@ RSAPublicKey publicKey() {
+                    if (!"https".equals(u.getScheme())) {
+                        return FormValidation.errorWithMarkup("Issuer URIs should use <code>https</code> scheme");
+                    }
+                    if (u.getQuery() != null) {
+                        return FormValidation.error("Issuer URIs must not have a query component");
+                    }
+                    if (u.getFragment() != null) {
+                        return FormValidation.error("Issuer URIs must not have a fragment component");
+                    }
+                    if (u.getPath() != null && u.getPath().endsWith("/")) {
+                        return FormValidation.errorWithMarkup("Issuer URIs should not end with a slash (<code>/</code>) in this context");
+                    }
+                } catch (URISyntaxException x) {
+                    return FormValidation.error("Not a well-formed URI");
+                }
+                if (i != null) {
+                    IdTokenCredentials c = i.credentials().stream().filter(creds -> creds.getId().equals(id) && issuer.equals(creds.getIssuer())).findFirst().orElse(null);
+                    if (c != null) {
+                        String base = req.getContextPath() + "/" + getDescriptorUrl();
+                        return FormValidation.okWithMarkup(
+                            "Serve <code>" + Util.xmlEscape(issuer) + Keys.WELL_KNOWN_OPENID_CONFIGURATION +
+                            "</code> with <a href=\"" + base + "/wellKnownOpenidConfiguration?issuer=" + Util.escape(issuer) +
+                            "\" target=\"_blank\" rel=\"noopener noreferrer\">this content</a> and <code>" +
+                            Util.xmlEscape(issuer) + Keys.JWKS + "</code> with <a href=\"" +
+                            base + "/jwks?uri=" + Util.escape(i.uri()) + "&id=" + Util.escape(id) + "&issuer=" + Util.escape(issuer) +
+                            "\" target=\"_blank\" rel=\"noopener noreferrer\">this content</a> (both as <code>application/json</code>)." +
+                            "<br>Note that the JWKS document will need to be updated if you resave these credentials.");
+                    } else {
+                        return FormValidation.ok("Save these credentials, then return to this screen for instructions");
+                    }
+                } else {
+                    return FormValidation.warning("Unable to determine where these credentials are being saved");
+                }
+            }
+        }
+
+        public JSONObject doWellKnownOpenidConfiguration(@QueryParameter String issuer) {
+            return Keys.openidConfiguration(issuer);
+        }
+
+        public JSONObject doJwks(@QueryParameter String uri, @QueryParameter String id, @QueryParameter String issuer) {
+            Issuer i = ExtensionList.lookup(Issuer.Factory.class).stream().map(f -> f.forUri(uri)).filter(Objects::nonNull).findFirst().orElse(null);
```

Comment:
Going to check whether I can replace the `uri` query parameter with a contextual lookup of the folder as in `doCheckIssuer`.